### PR TITLE
[HOLD] disable button on iou send page when offline

### PR DESCRIPTION
**Pull request by Justicea83**
### Fixed Issues

$https://github.com/Expensify/App/issues/8441

### Tests

1. https://staging.new.expensify.com
2. Log in with Account A in Windows/Chrome
3. Log in with Account B in any other device
4. Account B request ed money from Account A
5. Check if you see requested amount on Account A
6. Disable the internet connection
7. Click on Pay option
8. Verify that the `Pay With Expensify` or `I'll Settle Elsewhere` button is disabled in the screenshots below

</details>

### QA Steps

- [ ] Verify that no errors appear in the JS console

#### Web

Screenshot 2022-05-11 at 2 43 45 PM

#### Mobile Web

Screenshot 2022-05-11 at 2 59 26 PM

#### Desktop

Screenshot 2022-05-11 at 2 47 26 PM

#### iOS

Simulator Screen Shot - iPhone 13 - 2022-05-11 at 15 35 44

#### Android

Screenshot 2022-05-11 at 2 57 58 PM

### CODE REVIEW

**Review by amyevans — APPROVED** [review]
The code looks good to me, but since I'm still getting up to speed in this repo and @marcaaron is our Offline expert, I'd appreciate his eyes before we merge

**Comment by amyevans** [comment]
Oh, and 1 question for the room: are we trying to keep components up to date with new props in Storybook? Because if so, the ButtonWithDropdown could benefit from an update

**Comment by Justicea83** [comment]
Gentle bump

**Comment by marcaaron** [comment]
Thanks! I suspect @Gonals made this internal because it was not quite yet ready to be worked on. You're a little early @Justicea83, but no worries 😄 

Unsure if we need to block these changes (they're already done) - but the linked issue should technically be on `[HOLD]` as the proposed solution here isn't following the correct UX pattern. 

More insight in this design doc:

https://docs.google.com/document/d/1RWPLDgdrJ4eW1KfseVmrYYy_fDUWWDQpek-hmdzAtuM/edit#heading=h.ns12dr1f35rx

The correct pattern to use here would probably be B or C

To do B would be very complex for this particular flow. And to pull off C correctly we should be using the 

[`<OfflineIndicator />`](https://docs.google.com/document/d/1RWPLDgdrJ4eW1KfseVmrYYy_fDUWWDQpek-hmdzAtuM/edit#bookmark=id.1zhs12qgwg1d) 

component that has not yet been implemented.

And not the red text that is being rendered here:

https://github.com/Expensify/App/blob/11a1f0debac2dfee461e6b71751c3b3bcfb9f2cf/src/components/IOUConfirmationList.js#L382-L386

That said, I don't think it's a big deal if we merge this. But let's maybe wait for the Offline conversations to wrap up before we work on any more offline specific UX improvements like this.

----

>are we trying to keep components up to date with new props in Storybook? Because if so, the ButtonWithDropdown could benefit from an update

@amyevans never a bad thing to keep things up to date in SB, but it's not required (NAB) unless a change broke something or makes something go out of date there.

File: src/components/SettlementButton.js
Comment (marcaaron) [inline src/components/SettlementButton.js:135]: Ah this seems redundant in this usage here

https://github.com/Expensify/App/blob/11a1f0debac2dfee461e6b71751c3b3bcfb9f2cf/src/components/IOUConfirmationList.js#L387-L390

Because `shouldDisableButton` is already looking at `this.props.network.isOffline`

https://github.com/Expensify/App/blob/11a1f0debac2dfee461e6b71751c3b3bcfb9f2cf/src/components/IOUConfirmationList.js#L351-L352

**Review by marcaaron — CHANGES_REQUESTED** [review]

**Comment by Justicea83** [comment]
alright, thats fine, am going ahead and put this on hold. Thanks for the clarification @marcaaron .

**Comment by Justicea83** [comment]
waiting on the review of the IOU sub design doc

**Comment by Justicea83** [comment]
No overdue, just blocked atm

**Comment by Justicea83** [comment]
this is still blocked by the pattern B refactors

**Comment by Justicea83** [comment]
Stilling on holding pending pattern B

**Comment by Justicea83** [comment]
@marcaaron I think this PR doesn't hold again, no?

**Comment by Justicea83** [comment]
Closing this out as this is now part of the manual requests project(https://github.com/Expensify/App/issues/8441#issuecomment-1240875585)

**Pull request closed without merging**
